### PR TITLE
#UTCINT-110 added api doc for register user service terms action

### PR DESCRIPTION
**Pull request by harshilsharma**
Related to - https://github.com/mattermost/mattermost-server/pull/9410

### CODE REVIEW

File: v4/source/definitions.yaml [hunk 1]
```diff
@@ -2202,6 +2202,23 @@ definitions:
         type: string
         description: The id of the default channel user role for this scheme.
 
+  ServiceTerms:
```
Comment (jasonblais) [inline v4/source/definitions.yaml]: Propose the feature is referred to as "Terms Of Service" in all user-facing texts

Comment (jasonblais) [reply v4/source/definitions.yaml]: This can be `TermsOfService` for instance when no spaces are permitted.

Comment (harshilsharma63) [reply v4/source/definitions.yaml]: Yes, that is correct. Make the changes.

File: v4/source/definitions.yaml [hunk 2]
```diff
@@ -2202,6 +2202,23 @@ definitions:
         type: string
         description: The id of the default channel user role for this scheme.
 
+  TermsOfService:
+    type: object
+    properties:
+      id:
+        type: string
+        description: The unique identifier of the service terms.
```
Comment (jasonblais) [inline v4/source/definitions.yaml]: Would propose having `terms of service` here too instead of `service terms`

File: v4/source/definitions.yaml [hunk 3]
```diff
@@ -2202,6 +2202,23 @@ definitions:
...
+      create_at:
+        type: integer
+        format: int64
+        description: The time at which the service terms was created.
```
Comment (jasonblais) [inline v4/source/definitions.yaml]: Same here

File: v4/source/definitions.yaml [hunk 4]
```diff
@@ -2202,6 +2202,23 @@ definitions:
...
+      user_id:
+        type: string
+        description: The unique identifier of the user who created these service terms.
```
Comment (jasonblais) [inline v4/source/definitions.yaml]: Same here

File: v4/source/definitions.yaml [hunk 5]
```diff
@@ -2202,6 +2202,23 @@ definitions:
...
+      Text:
+        type: string
+        description: The text of service terms. Supports markdown.
```
Comment (jasonblais) [inline v4/source/definitions.yaml]: Same here

File: v4/source/service_terms.yaml [hunk 6]
```diff
@@ -0,0 +1,35 @@
+  /terms_of_service:
+    get:
+      tags:
+      - terms of service
+      summary: Get latest service terms
```
Comment (jasonblais) [inline v4/source/service_terms.yaml]: Same here (throughout the file)

File: v4/source/users.yaml [hunk 7]
```diff
@@ -1494,3 +1494,50 @@
           $ref: '#/responses/Forbidden'
         '501':
           $ref: '#/responses/NotImplemented'
+
+
+  /users/{user_id}/terms_of_service:
+    post:
+      tags:
+        - users
+      summary: Records user action when they accept or decline custom service terms
```
Comment (jasonblais) [inline v4/source/users.yaml]: Same here (throughout the file)

File: v4/source/definitions.yaml [hunk 8]
```diff
@@ -2202,6 +2202,23 @@ definitions:
...
+      create_at:
+        type: integer
+        format: int64
+        description: The time at which the terms of service was created.
+      user_id:
+        type: string
+        description: The unique identifier of the user who created these terms of service.
+      Text:
+        type: string
+        description: The text of terms of service. Supports markdown.
```
[the remark this answers is not in the document]
Comment (harshilsharma) [reply v4/source/definitions.yaml]: Done.

File: v4/source/users.yaml [hunk 9]
```diff
@@ -1494,3 +1494,50 @@
...
           $ref: '#/responses/NotImplemented'
+
+
+  /users/{user_id}/terms_of_service:
+    post:
+      tags:
+        - users
+      summary: Records user action when they accept or decline custom terms of service
+      description: |
+        Records user action when they accept or decline custom terms of service. Records the action in audot.
```
[the remark this answers is not in the document]
Comment (harshilsharma) [reply v4/source/users.yaml]: Done.

File: v4/source/users.yaml [hunk 10]
```diff
@@ -1494,3 +1494,50 @@
...
+        Update's user's last accepted terms of service ID if they accepted it.
```
[the remark this answers is not in the document]
Comment (harshilsharma) [reply v4/source/users.yaml]: Done.

File: v4/source/definitions.yaml [hunk 11]
```diff
@@ -2202,6 +2202,23 @@ definitions:
...
+        type: string
+        description: The unique identifier of the terms of service.
+      create_at:
+        type: integer
+        format: int64
+        description: The time at which the terms of service was created.
+      user_id:
+        type: string
+        description: The unique identifier of the user who created these terms of service.
+      Text:
```
Comment (grundleborg) [inline v4/source/definitions.yaml]: Should `Text` be all lower case?

File: v4/source/service_terms.yaml [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+        '401':
+          $ref: '#/responses/Unauthorized'
+    post:
+      tags:
+      - terms of service
+      summary: Creates a new terms of service
+      description: |
+        Creates new terms of service
+        ##### Permissions
+        Must be authenticated.
```
Comment (grundleborg) [inline v4/source/service_terms.yaml]: I don't think this is right. Shouldn't the user have `manage_system` permissions to create a ToS?

File: v4/source/service_terms.yaml [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+        '400':
+          $ref: '#/responses/BadRequest'
+        '401':
+          $ref: '#/responses/Unauthorized'
+    post:
+      tags:
+      - terms of service
+      summary: Creates a new terms of service
+      description: |
+        Creates new terms of service
```
Comment (grundleborg) [inline v4/source/service_terms.yaml]: Need a "minimum server version" here.

File: v4/source/service_terms.yaml [hunk 14]
```diff
@@ -0,0 +1,35 @@
+  /terms_of_service:
+    get:
+      tags:
+      - terms of service
+      summary: Get latest terms of service
+      description: |
+        Get latest terms of service from the server
+        ##### Permissions
+        Must be authenticated.
```
Comment (grundleborg) [inline v4/source/service_terms.yaml]: Need a "minimum server version" here.

File: v4/source/service_terms.yaml (hunk 13 above)
Comment (harshilsharma) [reply v4/source/service_terms.yaml]: Done.

File: v4/source/service_terms.yaml (hunk 14 above)
Comment (harshilsharma) [reply v4/source/service_terms.yaml]: Done.

File: v4/source/service_terms.yaml (hunk 12 above)
Comment (harshilsharma) [reply v4/source/service_terms.yaml]: Server PR - https://github.com/mattermost/mattermost-server/pull/9556

**Pull request merged**